From b1165d7dbc8d001024254d8d26bc740022f4d2a8 Mon Sep 17 00:00:00 2001
From: Jonas Bernoulli <jonas@bernoul.li>
Date: Sat, 19 Aug 2017 17:44:00 +0200
Subject: [PATCH] [PATCH] Add library flim.el whose name matches that of the
 package flim

A package should always contain a library with the same name.  One
benefit of that is that Melpa can then extract the package description
from the library commentary.

Remove "flim-pkg.el".  The only additional information it contains is a
version string, which has not ever been bumped since it was first added
in 2014.

[Emacsmirror] This commit is being rebased onto upstream.
---
 flim-pkg.el |  6 ------
 flim.el     | 34 ++++++++++++++++++++++++++++++++++
 2 files changed, 34 insertions(+), 6 deletions(-)
 delete mode 100644 flim-pkg.el
 create mode 100644 flim.el

diff --git a/flim-pkg.el b/flim-pkg.el
deleted file mode 100644
index 0c90ee8..0000000
--- a/flim-pkg.el
+++ /dev/null
@@ -1,6 +0,0 @@
-(define-package "flim" "1.14.9"
-  "A library to provide basic features about message representation or encoding."
-  '((emacs "24.5")
-    (apel "10.8")
-    (oauth2 "0.11"))
-  :maintainers '(("Kazuhiro Ito" . "kzhr@d1.dion.ne.jp")))
diff --git a/flim.el b/flim.el
new file mode 100644
index 0000000..cb64f7a
--- /dev/null
+++ b/flim.el
@@ -0,0 +1,34 @@
+;;; flim.el --- Basic message representation and encoding features
+
+;; Copyright (C) 1997-2024  Free Software Foundation, Inc.
+
+;; Keywords: MIME, multimedia, mail, news
+;; Package-Requires: ((emacs "24.5") (apel "0") (oauth2 "0.17"))
+
+;; This file is part of FLIM (Faithful Library about Internet Message).
+
+;; This program is free software; you can redistribute it and/or
+;; modify it under the terms of the GNU General Public License as
+;; published by the Free Software Foundation; either version 2, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs; see the file COPYING.  If not, write to the
+;; Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+;; Boston, MA 02110-1301, USA.
+
+;;; Commentary:
+
+;; FLIM (Faithful Library about Internet Message) provides basic
+;; message representation and encoding features.
+
+;;; Code:
+
+(provide 'flim)
+
+;;; flim.el ends here
